Use whisper-timestamp instead of timestamp & perf improvement

cammellos opened the pull request:
This commit does a few things:

==== Ordering of messages ====

Change the ordering of messages from a mixture of timestamp/clock-value to use
only clock-value.

Datemarks are now not used for sorting anymore, which means that the
order of messages is always causally related (not the case before, as we
were breaking this property by sorting by datemark), but datemark
calculation is unreliable (a reply to a message might have a timestamp <
then the message that is replied to).
So for timestamp calculation we
naively group them ignoring "out-of-order timestamp" messages, although
there's much to improve.
It fixes an issue whereby the user would change their time and the
message will be displayed in the past, although it is still possible to
craft a message with a lower clock value and order it in the past
(there's no way we can prevent this to some extent, but there are ways
to mitigate, but outside the scope of this PR).

We keep timestamp as a field as we don't want to completely rely on whisper for this.

==== Performance of receiving messages ====

The app would freeze on pulling messages from a mailserver (100 or so).
This is due to the JS Thread being hogged by CPU calculation, coupled
with the fact that we always tried to process messages all in one go.

This strategy can't scale, and given x is big enough (200,300,1000) the
UI will freeze.

Instead, each message is now processed separately, and we leave a gap
between processing each message for the UI to respond to user input
(otherwise the app freezes again).
Pulling messages will be longer overall, but the app will be usuable
while this happen (albeit it might slow down).
Other strategies are possible (calculate off-db and do a big swap,
avoiding many re-renders etc), but this is the reccommended strategy by
re-frame author (Solving the CPU Hog problem), so sounds like a safe
base point.

The underlying data structure for holding messages was also changed, we
used an immutable Red and Black Tree, same as a sorted map for clojure, but we use
a js library as is twice as performing then clojure sorted map.

We also don't sort messages again each time we receive them O(nlogn), but we
insert them in order O(logn).

Other data structures considered but discarded:
1) Plain vector, but performance prepending/insertion in the middle
(both O(n)) were not great, as not really suited for these operations.

2) Linked list, appealing as append/prepend is O(1), while insertion is
O(n). This is probably acceptable as messages tend to come in order
(from the db, so adding N messages is O(n)), or the network (most of
them prepends, or close to the head), while mailserver would not follow this path.
An implementation of a linked list was built, which performed roughtly the
same as a clojure sorted-map (although faster append/prepend), but not
worth the complexity of having our own implementation.

3) Clojure sorted-map, probably the most versatile, performance were
acceptable, but nowhere near the javascript implementation we decided on

4) Priority map, much slower than a sorted map (twice as slow)

5) Mutable sorted map, js implementation, (bintrees), not explored this very much, but from
just a quick benchmark, performance were much worse that clojure
immutable sorted map

Given that each message is now processed separately, saving the chat /
messages is also debounced to avoid spamming status-go with network
requests. This is a temporary measure for now until that's done directly
in status-go, without having to ping-pong with status-react.

Next steps performance wise is to move stuff to status-go, parsing of
transit, validation, which is heavy, at which point we can re-consider
performance and how to handle messages.

Fixes also an issue with the last message in the chat, we were using the
last message in the chat list, which might not necessarely be the last
message the chat has seen, in case messages were not loaded and a more
recent message is the database (say you fetch historical messages for
1-to-1 A, you don't have any messages in 1-to-1 chat B loaded, you receive an
historical message for chat B, it sets it as last message).

Also use clj beans instead of js->clj for type conversion

There's still a few things to do (update desktop dependencies), and I am still going through some regression tests, as the impact of the pr is quite large (sorry for the large pr, I can split it if you'd like me to do, but without an holistic approach is difficult to check whether it was worth it), but it's ready for review.


Fixes: #4193 #6946

cammellos committed:
Fix message ordering and improve performance rec. messages

This commit does a few things:

==== Ordering of messages ====

Change the ordering of messages from a mixture of timestamp/clock-value to use
only clock-value.

Datemarks are now not used for sorting anymore, which means that the
order of messages is always causally related (not the case before, as we
were breaking this property by sorting by datemark), but datemark
calculation is unreliable (a reply to a message might have a timestamp <
then the message that is replied to).
So for timestamp calculation we
naively group them ignoring "out-of-order timestamp" messages, although
there's much to improve.
It fixes an issue whereby the user would change their time and the
message will be displayed in the past, although it is still possible to
craft a message with a lower clock value and order it in the past
(there's no way we can prevent this to some extent, but there are ways
to mitigate, but outside the scope of this PR).

==== Performance of receiving messages ====

The app would freeze on pulling messages from a mailserver (100 or so).
This is due to the JS Thread being hogged by CPU calculation, coupled
with the fact that we always tried to process messages all in one go.

This strategy can't scale, and given x is big enough (200,300,1000) the
UI will freeze.

Instead, each message is now processed separately, and we leave a gap
between processing each message for the UI to respond to user input
(otherwise the app freezes again).
Pulling messages will be longer overall, but the app will be usuable
while this happen (albeit it might slow down).
Other strategies are possible (calculate off-db and do a big swap,
avoiding many re-renders etc), but this is the reccommended strategy by
re-frame author (Solving the CPU Hog problem), so sounds like a safe
base point.

The underlying data structure for holding messages was also changed, we
used an immutable Red and Black Tree, same as a sorted map for clojure, but we use
a js library as is twice as performing then clojure sorted map.

We also don't sort messages again each time we receive them O(nlogn), but we
insert them in order O(logn).

Other data structures considered but discarded:
1) Plain vector, but performance prepending/insertion in the middle
(both O(n)) were not great, as not really suited for these operations.

2) Linked list, appealing as append/prepend is O(1), while insertion is
O(n). This is probably acceptable as messages tend to come in order
(from the db, so adding N messages is O(n)), or the network (most of
them prepends, or close to the head), while mailserver would not follow this path.
An implementation of a linked list was built, which performed roughtly the
same as a clojure sorted-map (although faster append/prepend), but not
worth the complexity of having our own implementation.

3) Clojure sorted-map, probably the most versatile, performance were
acceptable, but nowhere near the javascript implementation we decided on

4) Priority map, much slower than a sorted map (twice as slow)

5) Mutable sorted map, js implementation, (bintrees), not explored this very much, but from
just a quick benchmark, performance were much worse that clojure
immutable sorted map

Given that each message is now processed separately, saving the chat /
messages is also debounced to avoid spamming status-go with network
requests. This is a temporary measure for now until that's done directly
in status-go, without having to ping-pong with status-react.

Next steps performance wise is to move stuff to status-go, parsing of
transit, validation, which is heavy, at which point we can re-consider
performance and how to handle messages.

Fixes also an issue with the last message in the chat, we were using the
last message in the chat list, which might not necessarely be the last
message the chat has seen, in case messages were not loaded and a more
recent message is the database (say you fetch historical messages for
1-to-1 A, you don't have any messages in 1-to-1 chat B loaded, you receive an
historical message for chat B, it sets it as last message).

Also use clj beans instead of js->clj for type conversion

Signed-off-by: Andrea Maria Piana <andrea.maria.piana@gmail.com>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 98947c8b | [#2](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-9294/2/) | 2019-10-27 20:57:19 | ~11 min | `ios` | [:package:`ipa`](https://i.diawi.com/NNL554) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FNNL554)|

Serhy commented:
@cammellos I checked this build with receiving message history with public chat having 100 messages, I also tried even with public having 1000 (in the past it was killing amount for Galaxy Note 4) and no lags when receiving messages. NO LAGS! **Less than 0.1 second on slow device**

Can navigate fine between screens fine disregard the number of messages are fetching from offline inbox.

This fix is mega promising!

cammellos commented:
Thanks @Serhy for the good news!

rasom commented:
@cammellos I see review is requested already, can you please add some notes on what happens here, PR is quite big

cammellos commented:
@rasom I think the review was requested automatically on creation of the PR (there must be some settings that someone has enabled), but it's not ready to be reviewed, I will add info in due time and re-request the review.
cheers

rasom commented:
@cammellos got it, thanks

cammellos commented:
@rasom ready to be reviewed now, thanks!

Serhy commented:
E2e tests failing above dues to PR build crash when message is received in 1-1 chat (all good for public chats).
I rechecked with real Android and indeed app crashes on message recipient side when any text message sent this contact (doesn't matter if recipient added in Contacts previously by sender or not)
And after that I can't login to account because it crashes all the time.

Logcat logs before message is received by contact:
``` 

11-01 10:47:28.999  7905  7972 E GoLog   : INFO [11-01|08:47:28.998] recovered installation                   namespace=Protocol site=recoverInstallationsFromBundle installation-id=b9ffd7fa-f8ac-56a3-914b-840f6ef091dc
11-01 10:47:29.009  7905  7972 E GoLog   : INFO [11-01|08:47:29.008] received a new installation              site=onNewInstallationsHandler identity=0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc id=b9ffd7fa-f8ac-56a3-914b-840f6ef091dc
11-01 10:47:30.083  7905  7963 D StatusModule: Signal event: {"type":"whisper.filter.added","event":{"filters":[{"chatId":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc-negotiated","symKeyId":"6d5088a2275e7aac564011c0646465e789d94dba28ed5cac7908a9af1eb36ed8","listen":true,"filterId":"92bda88e8545336d49e5c0d28c7b82b7c2000aec4cc2c11f6d9ca15685df1cae","identity":"04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","topic":"0x6b939797"}]}}
11-01 10:47:30.090  7905  7963 D StatusModule: Signal event: {"type":"messages.new","event":{"messages":[{"error":null,"messages":[{"message":{"sig":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","ttl":15,"timestamp":1572598040,"topic":"0xf8946aac","payload":"0x5b227e236334222c5b2262786e666a666a222c22746578742f706c61696e222c227e3a757365722d6d657373616765222c3135373235393830343034303130312c313537323539383034303430302c5b225e20222c227e3a636861742d6964222c22307830346337633766636336333437376233303338656262663761633530353763653966313631363234313535616561303939326639303134366231386431363230353334363531656635376163396236386266633864396565653561336637613432343264383662616137303062663235653233393033663231366662613366663935222c227e3a74657874222c2262786e666a666a225d5d5d","padding":"0xdb28b3ea3ae4669e76dac1ffcc17e97fa80f91f28a6fdfd795ba5a0ad318d1a5f6211e73b0b6d882ed89aa57471c28bfa923d92717d8a3a4a4983cf75c972d37350cfe862141dc2eaa0d55e2dd6746a15e71eb64a8b52e493313d0558f609a2e750a68c08b3cc26496c712a3ea88908eb675f4acf498ce33d7b6d4b9ba238a20fc418e","pow":0.018941916389197188,"hash":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","recipientPublicKey":"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95"},"metadata":{"dedupId":"ATIwMTkxMTAxYjFjNTAyYTgxYjM5MjM4YTQxYzk2N2MwYTg4MTY2ZWQxZDMwM2IwOTk4MDNkZWE1MWU1MzcwMDVmOTM5ZTc4Nth6WJ1nhGXGlq0o6STE9mvec7yxL3HtgnyzpApNV1jkhXZGnuH9UPns196yFDTi9ke5zBJWtUbdMLu8sARpEnc=","encryptionId":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","messageId":"0x3ae4ba175b7d61cb4dd4e34ebe1e1cef6eec17c794165b93089dfc1b3206d4d3","author":{"publicKey":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","alias":"Vital Crooked Myna","identicon":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAIAAACRXR/mAAAAhklEQVR4nOzXQQqAIBBA0QrP0gU9hxfsMraNCOub0Sz+W0bFZ0AGU611imf5O+CaWYRZhFmEWUSiH5StnJ7kNXe80xZ0WmYRZhFmEWYRZhFz++bzfrv1/SfotMwizCLMIswizCJuduITo/bmUdBpmUWYRQw4iV8IOi2zCLMIswiziD0AAP//OFwbWk+9t2sAAAAASUVORK5CYII="}},"payload":"[\"~#c4\",[\"bxnfjfj\",\"text/plain\",\"~:user-message\",157259804040101,1572598040400,[\"^ \",\"~:chat-id\",\"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95\",\"~:text\",\"bxnfjfj\"]]]"}],"chat":{"chatId":"contact-discovery","filterId":"ffe52d292c84fe6c5b87759cba50fcd9e437b4e5319b67b6cec84a4d06fc7d1f","symKeyId":"","oneToOne":true,"identity":"04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95","topic":"0xf8946aac","discovery":true,"negotiated":false,"listen":true}}]}}
11-01 10:47:30.090  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :signals/signal-received
11-01 10:47:30.094  7905  7958 D ReactNativeJS: DEBUG [status-im.events:659] - :event-str {"type":"whisper.filter.added","event":{"filters":[{"chatId":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc-negotiated","symKeyId":"6d5088a2275e7aac564011c0646465e789d94dba28ed5cac7908a9af1eb36ed8","listen":true,"filterId":"92bda88e8545336d49e5c0d28c7b82b7c2000aec4cc2c11f6d9ca15685df1cae","identity":"04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","topic":"0x6b939797"}]}}
11-01 10:47:30.099  7905  7958 D ReactNativeJS: DEBUG [status-im.transport.filters.core:330] - PERF :filters/add-raw-filters
11-01 10:47:30.105  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :filters.callback/filters-added
11-01 10:47:30.108  7905  7958 D ReactNativeJS: DEBUG [status-im.transport.filters.core:319] - PERF :filters.callback/filters-added
11-01 10:47:30.118  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:389] - Mailserver: planned requests  ({:topics #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"}, :from 1572598024, :to 1572598050, :force-to? false, :gap nil, :chat-id nil} {:topics #{"0x6b939797"}, :from 1572511650, :to 1572598050, :force-to? false, :gap nil, :chat-id nil})
11-01 10:47:30.126  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:263] - Adjusting mailserver request from: 1572598024 adjusted-from: 1572598004
11-01 10:47:30.130  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:294] - mailserver: request-messages for:   topics  #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"}  from  1572598004  force-to?  false  to  1572598050  range  26  cursor  nil  limit  2000
11-01 10:47:30.136  7905  8148 E GoLog   : INFO [11-01|08:47:30.135] RequestMessages                          package=status-go/services/sshext.PublicAPI request="{MailServerPeer:enode://74957e361ab290e6af45a124536bc9adee39fbd2f995a77ace6ed7d05d9a1c7c98b78b2df5f8071c439b9c0afe4a69893ede4ad633473f96bc195ddf33f6ce00@47.52.255.195:443 From:1572598004 To:0 Limit:2000 Cursor: Topic:[0 0 0 0] Topics:[[241 194 217 117] [153 173 6 55] [219 33 109 34] [248 148 106 172] [227 140 132 172]] SymKeyID:28e40a93a30bf03e1f9432fb66945ada98e268c4bb2a437d5921804b2a334b94 Timeout:30ns Force:false}"
11-01 10:47:30.138  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :signals/signal-received
11-01 10:47:30.142  7905  7958 D ReactNativeJS: DEBUG [status-im.events:659] - :event-str {"type":"messages.new","event":{"messages":[{"error":null,"messages":[{"message":{"sig":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","ttl":15,"timestamp":1572598040,"topic":"0xf8946aac","payload":"0x5b227e236334222c5b2262786e666a666a222c22746578742f706c61696e222c227e3a757365722d6d657373616765222c3135373235393830343034303130312c313537323539383034303430302c5b225e20222c227e3a636861742d6964222c22307830346337633766636336333437376233303338656262663761633530353763653966313631363234313535616561303939326639303134366231386431363230353334363531656635376163396236386266633864396565653561336637613432343264383662616137303062663235653233393033663231366662613366663935222c227e3a74657874222c2262786e666a666a225d5d5d","padding":"0xdb28b3ea3ae4669e76dac1ffcc17e97fa80f91f28a6fdfd795ba5a0ad318d1a5f6211e73b0b6d882ed89aa57471c28bfa923d92717d8a3a4a4983cf75c972d37350cfe862141dc2eaa0d55e2dd6746a15e71eb64a8b52e493313d0558f609a2e750a68c08b3cc26496c712a3ea88908eb675f4acf498ce33d7b6d4b9ba238a20fc418e","pow":0.018941916389197188,"hash":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","recipientPublicKey":"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95"},"metadata":{"dedupId":"ATIwMTkxMTAxYjFjNTAyYTgxYjM5MjM4YTQxYzk2N2MwYTg4MTY2ZWQxZDMwM2IwOTk4MDNkZWE1MWU1MzcwMDVmOTM5ZTc4Nth6WJ1nhGXGlq0o6STE9mvec7yxL3HtgnyzpApNV1jkhXZGnuH9UPns196yFDTi9ke5zBJWtUbdMLu8sARpEnc=","encryptionId":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","messageId":"0x3ae4ba175b7d61cb4dd4e34ebe1e1cef6eec17c794165b93089dfc1b3206d4d3","author":{"publicKey":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","alias":"Vital Crooked Myna","identicon":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAIAAACRXR/mAAAAhklEQVR4nOzXQQqAIBBA0QrP0gU9hxfsMraNCOub0Sz+W0bFZ0AGU611imf5O+CaWYRZhFmEWUSiH5StnJ7kNXe80xZ0WmYRZhFmEWYRZhFz++bzfrv1/SfotMwizCLMIswizCJuduITo/bmUdBpmUWYRQw4iV8IOi2zCLMIswiziD0AAP//OFwbWk+9t2sAAAAASUVORK5CYII="}},"payload":"[\"~#c4\",[\"bxnfjfj\",\"text/plain\",\"~:user-message\",157259804040101,1572598040400,[\"^ \",\"~:chat-id\",\"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95\",\"~:text\",\"bxnfjfj\"]]]"}],"chat":{"chatId":"contact-discovery","filterId":"ffe52d292c84fe6c5b87759cba50fcd9e437b4e5319b67b6cec84a4d06fc7d1f","symKeyId":"","oneToOne":true,"identity":"04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95","topic":"0xf8946aac","discovery":true,"negotiated":false,"listen":true}}]}}
11-01 10:47:30.149  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :status-im.transport.message.core/process
11-01 10:47:30.168  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.topics:36] - added mailserver-topic successfully
11-01 10:47:30.173  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:313] - mailserver: messages request success for topic  #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"} from 1572598024 to 1572598050
11-01 10:47:30.178  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :mailserver.callback/request-success
11-01 10:47:30.499  7905  8159 D StatusModule: Signal event: {"type":"mailserver.request.completed","event":{"requestID":"0x7f01277c7ddad2ee338e6a56da34609db86f4759bc7fefc37cc684623239ff63","lastEnvelopeHash":"0xa8241870f40c27e9a6ad1ae5269156b534b7cf567e2dda2db96e8574d87d658b","cursor":"","errorMessage":""}}
11-01 10:47:30.509  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :signals/signal-received
11-01 10:47:30.516  7905  7958 D ReactNativeJS: DEBUG [status-im.events:659] - :event-str {"type":"mailserver.request.completed","event":{"requestID":"0x7f01277c7ddad2ee338e6a56da34609db86f4759bc7fefc37cc684623239ff63","lastEnvelopeHash":"0xa8241870f40c27e9a6ad1ae5269156b534b7cf567e2dda2db96e8574d87d658b","cursor":"","errorMessage":""}}
11-01 10:47:30.535  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:716] - mailserver: message request  0x7f01277c7ddad2ee338e6a56da34609db86f4759bc7fefc37cc684623239ff63 completed for mailserver topics #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"} from 1572598024 to 1572598050
11-01 10:47:30.544  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:389] - Mailserver: planned requests  ({:topics #{"0x6b939797"}, :from 1572511650, :to 1572598050, :force-to? false, :gap nil, :chat-id nil})
11-01 10:47:30.566  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:263] - Adjusting mailserver request from: 1572511650 adjusted-from: 1572511630
11-01 10:47:30.571  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:294] - mailserver: request-messages for:   topics  #{"0x6b939797"}  from  1572511630  force-to?  false  to  1572598050  range  86400  cursor  nil  limit  2000
11-01 10:47:30.578  7905  7958 I ReactNativeJS: INFO [status-im.data-store.mailservers:35] - deleted gaps successfully
11-01 10:47:30.578  7905  8148 E GoLog   : INFO [11-01|08:47:30.577] RequestMessages                          package=status-go/services/sshext.PublicAPI request="{MailServerPeer:enode://74957e361ab290e6af45a124536bc9adee39fbd2f995a77ace6ed7d05d9a1c7c98b78b2df5f8071c439b9c0afe4a69893ede4ad633473f96bc195ddf33f6ce00@47.52.255.195:443 From:1572511630 To:0 Limit:2000 Cursor: Topic:[0 0 0 0] Topics:[[107 147 151 151]] SymKeyID:28e40a93a30bf03e1f9432fb66945ada98e268c4bb2a437d5921804b2a334b94 Timeout:30ns Force:false}"
11-01 10:47:30.582  7905  7958 I ReactNativeJS: INFO [status-im.data-store.mailservers:28] - saved gaps successfully
11-01 10:47:30.587  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:750] - added mailserver-topic successfully
11-01 10:47:30.602  7905  7958 I chatty  : uid=11648(im.status.ethereum.pr) mqt_js identical 3 lines
11-01 10:47:30.607  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:750] - added mailserver-topic successfully
11-01 10:47:30.611  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:313] - mailserver: messages request success for topic  #{"0x6b939797"} from 1572511650 to 1572598050
11-01 10:47:30.616  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :mailserver.callback/request-success
11-01 10:47:30.677  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :status-im.data-store.messages/save-messages
11-01 10:47:30.687  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :status-im.chat.models/save-chat
11-01 10:47:30.696  7905  8148 E GoLog   : INFO [11-01|08:47:30.696] saving chat                              package=status-go/services/sshext.PublicAPI chat="{ID: Name: Color: Active:true ChatType:1 PublicKey:<nil> Timestamp:0 LastClockValue:157259804040101 DeletedAtClockValue:0 UnviewedMessagesCount:1 LastMessageContentType:text/plain LastMessageContent:{\"chat-id\":\"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95\",\"text\":\"bxnfjfj\"} LastMessageTimestamp:1572598040400 Members:[] MembershipUpdates:[]}"
11-01 10:47:30.699  7905     0 E Go      : panic: runtime error: slice bounds out of range
11-01 10:47:30.699  7905     0 E Go      : 
11-01 10:47:30.699  7905     0 E Go      : goroutine 5473 [running]:
11-01 10:47:30.699  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go.sqlitePersistence.SaveChat(0x9c2a2180, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x1, 0x0, ...)
11-01 10:47:30.699  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go/persistence.go:52 +0x6bc
11-01 10:47:30.699  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go.(*Messenger).SaveChat(...)
11-01 10:47:30.699  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go/messenger.go:475
11-01 10:47:30.699  7905     0 E Go      : github.com/status-im/status-go/services/shhext.(*PublicAPI).SaveChat(0x9c08aa60, 0xcde63820, 0x9c2ec640, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, ...)
11-01 10:47:30.699  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/services/shhext/api.go:571 +0x124
11-01 10:47:30.699  7905     0 E Go      : reflect.Value.call(0x9c21c700, 0x9c2ef758, 0x13, 0xcd4f1a79, 0x4, 0x9e36aba0, 0x3, 0x4, 0x0, 0x9e36aba0, ...)
11-01 10:47:30.699  7905     0 E Go      : 	/nix/store/0la4ksmvi6s6j3bp7hw9jk4fji9w0ym1-go-1.12.9/share/go/src/reflect/value.go:447 +0x3b0
11-01 10:47:30.700  7905     0 E Go      : reflect.Value.Call(0x9c21c700, 0x9c2ef758, 0x13, 0x9e36aba0, 0x3, 0x4, 0x1, 0x1, 0x0)
11-01 10:47:30.700  7905     0 E Go      : 	/nix/store/0la4ksmvi6s6j3bp7hw9jk4fji9w0ym1-go-1.12.9/share/go/src/reflect/value.go:308 +0x78
11-01 10:47:30.700  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).handle(0x9c085760, 0xcde63820, 0x9c2ec640, 0xcde69358, 0x9c426a20, 0x9e36ab70, 0x0, 0xcccc0ae8, 0x4)
11-01 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:309 +0x5c0
11-01 10:47:30.700  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).exec(0x9c085760, 0xcde63820, 0x9c2ec640, 0xcde69358, 0x9c426a20, 0x9e36ab70)
11-01 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:330 +0x184
11-01 10:47:30.700  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest.func2(0x9c2f2040, 0x9c085760, 0x9c256088, 0xcde69358, 0x9c426a20, 0x9d852808, 0x1, 0x1, 0x0)
11-01 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:204 +0x8c
11-01 10:47:30.700  7905     0 E Go      : created by github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest
11-01 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:199 +0x210
--------- beginning of crash
11-01 10:47:30.700  7905  8159 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 8159 (Thread-11), pid 7905 (tus.ethereum.pr)

```

cammellos commented:
that s super helpful, thanks!

On Fri, Nov 1, 2019, 09:52 Sergey Chumak <notifications@github.com> wrote:

> E2e tests failing above dues to PR build crash when message is received in
> 1-1 chat (all good for public chats).
> I rechecked with real Android and indeed app crashes on message recipient
> side when any text message sent this contact (doesn't matter if recipient
> added in Contacts previously by sender or not)
> And after that I can't login to account because it crashes all the time.
>
> Logcat logs before message is received by contact:
>
>
> 11-01 10:47:28.999  7905  7972 E GoLog   : INFO [11-01|08:47:28.998] recovered installation                   namespace=Protocol site=recoverInstallationsFromBundle installation-id=b9ffd7fa-f8ac-56a3-914b-840f6ef091dc
> 11-01 10:47:29.009  7905  7972 E GoLog   : INFO [11-01|08:47:29.008] received a new installation              site=onNewInstallationsHandler identity=0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc id=b9ffd7fa-f8ac-56a3-914b-840f6ef091dc
> 11-01 10:47:30.083  7905  7963 D StatusModule: Signal event: {"type":"whisper.filter.added","event":{"filters":[{"chatId":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc-negotiated","symKeyId":"6d5088a2275e7aac564011c0646465e789d94dba28ed5cac7908a9af1eb36ed8","listen":true,"filterId":"92bda88e8545336d49e5c0d28c7b82b7c2000aec4cc2c11f6d9ca15685df1cae","identity":"04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","topic":"0x6b939797"}]}}
> 11-01 10:47:30.090  7905  7963 D StatusModule: Signal event: {"type":"messages.new","event":{"messages":[{"error":null,"messages":[{"message":{"sig":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","ttl":15,"timestamp":1572598040,"topic":"0xf8946aac","payload":"0x5b227e236334222c5b2262786e666a666a222c22746578742f706c61696e222c227e3a757365722d6d657373616765222c3135373235393830343034303130312c313537323539383034303430302c5b225e20222c227e3a636861742d6964222c22307830346337633766636336333437376233303338656262663761633530353763653966313631363234313535616561303939326639303134366231386431363230353334363531656635376163396236386266633864396565653561336637613432343264383662616137303062663235653233393033663231366662613366663935222c227e3a74657874222c2262786e666a666a225d5d5d","padding":"0xdb28b3ea3ae4669e76dac1ffcc17e97fa80f91f28a6fdfd795ba5a0ad318d1a5f6211e73b0b6d882ed89aa57471c28bfa923d92717d8a3a4a4983cf75c972d37350cfe862141dc2eaa0d55e2dd6746a15e71eb64a8b52e493313d0558f609a2e750a68c08b3cc26496c712a3ea88908eb675f4acf498ce33d7b6d4b9ba238a20fc418e","pow":0.018941916389197188,"hash":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","recipientPublicKey":"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95"},"metadata":{"dedupId":"ATIwMTkxMTAxYjFjNTAyYTgxYjM5MjM4YTQxYzk2N2MwYTg4MTY2ZWQxZDMwM2IwOTk4MDNkZWE1MWU1MzcwMDVmOTM5ZTc4Nth6WJ1nhGXGlq0o6STE9mvec7yxL3HtgnyzpApNV1jkhXZGnuH9UPns196yFDTi9ke5zBJWtUbdMLu8sARpEnc=","encryptionId":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","messageId":"0x3ae4ba175b7d61cb4dd4e34ebe1e1cef6eec17c794165b93089dfc1b3206d4d3","author":{"publicKey":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","alias":"Vital Crooked Myna","identicon":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAIAAACRXR/mAAAAhklEQVR4nOzXQQqAIBBA0QrP0gU9hxfsMraNCOub0Sz+W0bFZ0AGU611imf5O+CaWYRZhFmEWUSiH5StnJ7kNXe80xZ0WmYRZhFmEWYRZhFz++bzfrv1/SfotMwizCLMIswizCJuduITo/bmUdBpmUWYRQw4iV8IOi2zCLMIswiziD0AAP//OFwbWk+9t2sAAAAASUVORK5CYII="}},"payload":"[\"~#c4\",[\"bxnfjfj\",\"text/plain\",\"~:user-message\",157259804040101,1572598040400,[\"^ \",\"~:chat-id\",\"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95\",\"~:text\",\"bxnfjfj\"]]]"}],"chat":{"chatId":"contact-discovery","filterId":"ffe52d292c84fe6c5b87759cba50fcd9e437b4e5319b67b6cec84a4d06fc7d1f","symKeyId":"","oneToOne":true,"identity":"04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95","topic":"0xf8946aac","discovery":true,"negotiated":false,"listen":true}}]}}
> 11-01 10:47:30.090  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :signals/signal-received
> 11-01 10:47:30.094  7905  7958 D ReactNativeJS: DEBUG [status-im.events:659] - :event-str {"type":"whisper.filter.added","event":{"filters":[{"chatId":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc-negotiated","symKeyId":"6d5088a2275e7aac564011c0646465e789d94dba28ed5cac7908a9af1eb36ed8","listen":true,"filterId":"92bda88e8545336d49e5c0d28c7b82b7c2000aec4cc2c11f6d9ca15685df1cae","identity":"04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","topic":"0x6b939797"}]}}
> 11-01 10:47:30.099  7905  7958 D ReactNativeJS: DEBUG [status-im.transport.filters.core:330] - PERF :filters/add-raw-filters
> 11-01 10:47:30.105  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :filters.callback/filters-added
> 11-01 10:47:30.108  7905  7958 D ReactNativeJS: DEBUG [status-im.transport.filters.core:319] - PERF :filters.callback/filters-added
> 11-01 10:47:30.118  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:389] - Mailserver: planned requests  ({:topics #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"}, :from 1572598024, :to 1572598050, :force-to? false, :gap nil, :chat-id nil} {:topics #{"0x6b939797"}, :from 1572511650, :to 1572598050, :force-to? false, :gap nil, :chat-id nil})
> 11-01 10:47:30.126  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:263] - Adjusting mailserver request from: 1572598024 adjusted-from: 1572598004
> 11-01 10:47:30.130  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:294] - mailserver: request-messages for:   topics  #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"}  from  1572598004  force-to?  false  to  1572598050  range  26  cursor  nil  limit  2000
> 11-01 10:47:30.136  7905  8148 E GoLog   : INFO [11-01|08:47:30.135] RequestMessages                          package=status-go/services/sshext.PublicAPI request="{MailServerPeer:enode://74957e361ab290e6af45a124536bc9adee39fbd2f995a77ace6ed7d05d9a1c7c98b78b2df5f8071c439b9c0afe4a69893ede4ad633473f96bc195ddf33f6ce00@47.52.255.195:443 From:1572598004 To:0 Limit:2000 Cursor: Topic:[0 0 0 0] Topics:[[241 194 217 117] [153 173 6 55] [219 33 109 34] [248 148 106 172] [227 140 132 172]] SymKeyID:28e40a93a30bf03e1f9432fb66945ada98e268c4bb2a437d5921804b2a334b94 Timeout:30ns Force:false}"
> 11-01 10:47:30.138  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :signals/signal-received
> 11-01 10:47:30.142  7905  7958 D ReactNativeJS: DEBUG [status-im.events:659] - :event-str {"type":"messages.new","event":{"messages":[{"error":null,"messages":[{"message":{"sig":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","ttl":15,"timestamp":1572598040,"topic":"0xf8946aac","payload":"0x5b227e236334222c5b2262786e666a666a222c22746578742f706c61696e222c227e3a757365722d6d657373616765222c3135373235393830343034303130312c313537323539383034303430302c5b225e20222c227e3a636861742d6964222c22307830346337633766636336333437376233303338656262663761633530353763653966313631363234313535616561303939326639303134366231386431363230353334363531656635376163396236386266633864396565653561336637613432343264383662616137303062663235653233393033663231366662613366663935222c227e3a74657874222c2262786e666a666a225d5d5d","padding":"0xdb28b3ea3ae4669e76dac1ffcc17e97fa80f91f28a6fdfd795ba5a0ad318d1a5f6211e73b0b6d882ed89aa57471c28bfa923d92717d8a3a4a4983cf75c972d37350cfe862141dc2eaa0d55e2dd6746a15e71eb64a8b52e493313d0558f609a2e750a68c08b3cc26496c712a3ea88908eb675f4acf498ce33d7b6d4b9ba238a20fc418e","pow":0.018941916389197188,"hash":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","recipientPublicKey":"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95"},"metadata":{"dedupId":"ATIwMTkxMTAxYjFjNTAyYTgxYjM5MjM4YTQxYzk2N2MwYTg4MTY2ZWQxZDMwM2IwOTk4MDNkZWE1MWU1MzcwMDVmOTM5ZTc4Nth6WJ1nhGXGlq0o6STE9mvec7yxL3HtgnyzpApNV1jkhXZGnuH9UPns196yFDTi9ke5zBJWtUbdMLu8sARpEnc=","encryptionId":"0x2772812b288b5f49e0a0fc321272162403172f1f3c747947632272aafae8318e","messageId":"0x3ae4ba175b7d61cb4dd4e34ebe1e1cef6eec17c794165b93089dfc1b3206d4d3","author":{"publicKey":"0x04a9a904cd0d84674e165fa77ff5519bf9b84e8a314e4f21dcbbfaafce667b34604d1aef8db3860897d44519395571f5c42f599e5a6e2a4320c1020913ac14b4bc","alias":"Vital Crooked Myna","identicon":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAIAAACRXR/mAAAAhklEQVR4nOzXQQqAIBBA0QrP0gU9hxfsMraNCOub0Sz+W0bFZ0AGU611imf5O+CaWYRZhFmEWUSiH5StnJ7kNXe80xZ0WmYRZhFmEWYRZhFz++bzfrv1/SfotMwizCLMIswizCJuduITo/bmUdBpmUWYRQw4iV8IOi2zCLMIswiziD0AAP//OFwbWk+9t2sAAAAASUVORK5CYII="}},"payload":"[\"~#c4\",[\"bxnfjfj\",\"text/plain\",\"~:user-message\",157259804040101,1572598040400,[\"^ \",\"~:chat-id\",\"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95\",\"~:text\",\"bxnfjfj\"]]]"}],"chat":{"chatId":"contact-discovery","filterId":"ffe52d292c84fe6c5b87759cba50fcd9e437b4e5319b67b6cec84a4d06fc7d1f","symKeyId":"","oneToOne":true,"identity":"04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95","topic":"0xf8946aac","discovery":true,"negotiated":false,"listen":true}}]}}
> 11-01 10:47:30.149  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :status-im.transport.message.core/process
> 11-01 10:47:30.168  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.topics:36] - added mailserver-topic successfully
> 11-01 10:47:30.173  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:313] - mailserver: messages request success for topic  #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"} from 1572598024 to 1572598050
> 11-01 10:47:30.178  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :mailserver.callback/request-success
> 11-01 10:47:30.499  7905  8159 D StatusModule: Signal event: {"type":"mailserver.request.completed","event":{"requestID":"0x7f01277c7ddad2ee338e6a56da34609db86f4759bc7fefc37cc684623239ff63","lastEnvelopeHash":"0xa8241870f40c27e9a6ad1ae5269156b534b7cf567e2dda2db96e8574d87d658b","cursor":"","errorMessage":""}}
> 11-01 10:47:30.509  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :signals/signal-received
> 11-01 10:47:30.516  7905  7958 D ReactNativeJS: DEBUG [status-im.events:659] - :event-str {"type":"mailserver.request.completed","event":{"requestID":"0x7f01277c7ddad2ee338e6a56da34609db86f4759bc7fefc37cc684623239ff63","lastEnvelopeHash":"0xa8241870f40c27e9a6ad1ae5269156b534b7cf567e2dda2db96e8574d87d658b","cursor":"","errorMessage":""}}
> 11-01 10:47:30.535  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:716] - mailserver: message request  0x7f01277c7ddad2ee338e6a56da34609db86f4759bc7fefc37cc684623239ff63 completed for mailserver topics #{"0xf1c2d975" "0x99ad0637" "0xdb216d22" "0xf8946aac" "0xe38c84ac"} from 1572598024 to 1572598050
> 11-01 10:47:30.544  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:389] - Mailserver: planned requests  ({:topics #{"0x6b939797"}, :from 1572511650, :to 1572598050, :force-to? false, :gap nil, :chat-id nil})
> 11-01 10:47:30.566  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:263] - Adjusting mailserver request from: 1572511650 adjusted-from: 1572511630
> 11-01 10:47:30.571  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:294] - mailserver: request-messages for:   topics  #{"0x6b939797"}  from  1572511630  force-to?  false  to  1572598050  range  86400  cursor  nil  limit  2000
> 11-01 10:47:30.578  7905  7958 I ReactNativeJS: INFO [status-im.data-store.mailservers:35] - deleted gaps successfully
> 11-01 10:47:30.578  7905  8148 E GoLog   : INFO [11-01|08:47:30.577] RequestMessages                          package=status-go/services/sshext.PublicAPI request="{MailServerPeer:enode://74957e361ab290e6af45a124536bc9adee39fbd2f995a77ace6ed7d05d9a1c7c98b78b2df5f8071c439b9c0afe4a69893ede4ad633473f96bc195ddf33f6ce00@47.52.255.195:443 From:1572511630 To:0 Limit:2000 Cursor: Topic:[0 0 0 0] Topics:[[107 147 151 151]] SymKeyID:28e40a93a30bf03e1f9432fb66945ada98e268c4bb2a437d5921804b2a334b94 Timeout:30ns Force:false}"
> 11-01 10:47:30.582  7905  7958 I ReactNativeJS: INFO [status-im.data-store.mailservers:28] - saved gaps successfully
> 11-01 10:47:30.587  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:750] - added mailserver-topic successfully
> 11-01 10:47:30.602  7905  7958 I chatty  : uid=11648(im.status.ethereum.pr) mqt_js identical 3 lines
> 11-01 10:47:30.607  7905  7958 D ReactNativeJS: DEBUG [status-im.mailserver.core:750] - added mailserver-topic successfully
> 11-01 10:47:30.611  7905  7958 I ReactNativeJS: INFO [status-im.mailserver.core:313] - mailserver: messages request success for topic  #{"0x6b939797"} from 1572511650 to 1572598050
> 11-01 10:47:30.616  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :mailserver.callback/request-success
> 11-01 10:47:30.677  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :status-im.data-store.messages/save-messages
> 11-01 10:47:30.687  7905  7958 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :status-im.chat.models/save-chat
> 11-01 10:47:30.696  7905  8148 E GoLog   : INFO [11-01|08:47:30.696] saving chat                              package=status-go/services/sshext.PublicAPI chat="{ID: Name: Color: Active:true ChatType:1 PublicKey:<nil> Timestamp:0 LastClockValue:157259804040101 DeletedAtClockValue:0 UnviewedMessagesCount:1 LastMessageContentType:text/plain LastMessageContent:{\"chat-id\":\"0x04c7c7fcc63477b3038ebbf7ac5057ce9f161624155aea0992f90146b18d1620534651ef57ac9b68bfc8d9eee5a3f7a4242d86baa700bf25e23903f216fba3ff95\",\"text\":\"bxnfjfj\"} LastMessageTimestamp:1572598040400 Members:[] MembershipUpdates:[]}"
> 11-01 10:47:30.699  7905     0 E Go      : panic: runtime error: slice bounds out of range
> 11-01 10:47:30.699  7905     0 E Go      :
> 11-01 10:47:30.699  7905     0 E Go      : goroutine 5473 [running]:
> 11-01 10:47:30.699  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go.sqlitePersistence.SaveChat(0x9c2a2180, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x1, 0x0, ...)
> 11-01 10:47:30.699  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go/persistence.go:52 +0x6bc
> 11-01 10:47:30.699  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go.(*Messenger).SaveChat(...)
> 11-01 10:47:30.699  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go/messenger.go:475
> 11-01 <http://github.com/status-im/status-go/vendor/github.com/status-im/status-protocol-go/messenger.go:47511-01> 10:47:30.699  7905     0 E Go      : github.com/status-im/status-go/services/shhext.(*PublicAPI).SaveChat(0x9c08aa60, 0xcde63820, 0x9c2ec640, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, ...)
> 11-01 10:47:30.699  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/services/shhext/api.go:571 +0x124
> 11-01 10:47:30.699  7905     0 E Go      : reflect.Value.call(0x9c21c700, 0x9c2ef758, 0x13, 0xcd4f1a79, 0x4, 0x9e36aba0, 0x3, 0x4, 0x0, 0x9e36aba0, ...)
> 11-01 10:47:30.699  7905     0 E Go      : 	/nix/store/0la4ksmvi6s6j3bp7hw9jk4fji9w0ym1-go-1.12.9/share/go/src/reflect/value.go:447 +0x3b0
> 11-01 10:47:30.700  7905     0 E Go      : reflect.Value.Call(0x9c21c700, 0x9c2ef758, 0x13, 0x9e36aba0, 0x3, 0x4, 0x1, 0x1, 0x0)
> 11-01 10:47:30.700  7905     0 E Go      : 	/nix/store/0la4ksmvi6s6j3bp7hw9jk4fji9w0ym1-go-1.12.9/share/go/src/reflect/value.go:308 +0x78
> 11-01 10:47:30.700  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).handle(0x9c085760, 0xcde63820, 0x9c2ec640, 0xcde69358, 0x9c426a20, 0x9e36ab70, 0x0, 0xcccc0ae8, 0x4)
> 11-01 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:309 +0x5c0
> 11-01 10:47:30.700  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).exec(0x9c085760, 0xcde63820, 0x9c2ec640, 0xcde69358, 0x9c426a20, 0x9e36ab70)
> 11-01 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:330 +0x184
> 11-01 10:47:30.700  7905     0 E Go      : github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest.func2(0x9c2f2040, 0x9c085760, 0x9c256088, 0xcde69358, 0x9c426a20, 0x9d852808, 0x1, 0x1, 0x0)
> 11-01 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:204 +0x8c
> 11-01 10:47:30.700  7905     0 E Go      : created by github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest
> 11-01 <http://github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest11-01> 10:47:30.700  7905     0 E Go      : 	/build/go/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/rpc/server.go:199 +0x210
> --------- beginning of crash
> 11-01 10:47:30.700  7905  8159 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 8159 (Thread-11), pid 7905 (tus.ethereum.pr)
>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/status-im/status-react/pull/9294?email_source=notifications&email_token=AAHYJMAV3PFV4U54RY7KX7TQRPU3PA5CNFSM4JFTKHFKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEC2LHDA#issuecomment-548713356>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHYJMHNIFYWBKEZ6AOGU5DQRPU3PANCNFSM4JFTKHFA>
> .
>

rasom commented on src/status_im/data_store/messages.cljs:
```diff
@@ -109,8 +109,24 @@
+  (when-let [messages (vals (:messages/stored db))]
+    ;; Pull message from database to pick up most recent changes, default to
+    ;; stored one in case it has been offloaded
+    (let [hydrated-messages (map #(get-in db [:chats (-> % :content :chat-id) :messages (:message-id %)] %) messages)]
```
a bit hard to read that map's function

cammellos replied:
I'll make it prettier

rasom commented on src/status_im/utils/types.cljs:
```diff
@@ -1,13 +1,15 @@
     (name s)
     s))
 
 (defn clj->json [data]
```
Do we need to rename this function or it will work with clojure data structures as well?

cammellos replied:
I think it just works, but will double check

rasom commented on src/status_im/transport/message/core.cljs:
```diff
@@ -64,36 +61,42 @@
+                              messages)
+                      acc)))
+                []
+                (.-messages event-js))]
```
Can we use beans here as well?

cammellos replied:
beans is much slower then pure javascript from our tests, so I would avoid in critical paths , but I can double check if that's still the case

rasom replied:
ok, got it

cammellos commented:
@jakubgs @vkjr I am getting an error on CI for desktop builds:

```
Unable to resolve module `scheduler` from `/home/jenkins/workspace/status-react_prs_linux_PR-9294/node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-prod.js`: Module `scheduler` does not exist in the Haste module map
```

I have added a dependency, but otherwise left them unchanged (I believe scheduler was there already), any clue?

vkjr commented:
@cammellos, I see that scheduler was removed in `desktop/js_files/yarn.lock`, whereas your error message says it is used in `react-native-desktop`. I suspect this is a reason of a problem.
And `scheduler` removed because `react` package version changed to `16.11` which I suspect doesn't rely on this package. Probably version of `react` can be reverted.

`react-native-desktop` derived from version 0.57.8 of react-native so I would suggest not to update data in `yarn.lock` synchronously for mobile and desktop builds because mobile is at higher `react-native` version.

cammellos commented:
@vkjr thanks, I'll look into it!

cammellos commented:
@Serhy @churik I will be merging this as I have run the tests against develop and these 3 tests fail as well, I will take a look at the offline test.
